tests/components/pushover/test_notify.py

```diff
@@ -0,0 +1,150 @@
+    hass: HomeAssistant, mock_pushover: MagicMock
+) -> None:
+    """Test pushover failed setup due to invalid api key."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_CONFIG,
+    )
+    entry.add_to_hass(hass)
+    mock_pushover.side_effect = BadAPIRequestError("400: application token is invalid")
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+    assert entry.state is ConfigEntryState.SETUP_ERROR
+
+
+async def test_async_setup_entry_failed_conn_error(
+    hass: HomeAssistant, mock_pushover: MagicMock
+) -> None:
+    """Test pushover failed setup due to conn error."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_CONFIG,
+    )
+    entry.add_to_hass(hass)
+    mock_pushover.side_effect = BadAPIRequestError
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+    assert entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_async_setup_entry_failed_json_error(
+    hass: HomeAssistant, requests_mock: requests_mock.Mocker
+) -> None:
+    """Test pushover failed setup due to bad json response from library."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_CONFIG,
+    )
+    entry.add_to_hass(hass)
+    requests_mock.post(
+        "https://api.pushover.net/1/users/validate.json", status_code=204
+    )
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+    assert entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_async_setup_entry_failed_urrlib3_error(
+    hass: HomeAssistant, mock_pushover: MagicMock
+) -> None:
+    """Test pushover failed setup due to conn error."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=MOCK_CONFIG,
+    )
+    entry.add_to_hass(hass)
+    mock_pushover.side_effect = MaxRetryError(MagicMock(), MagicMock())
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+    assert entry.state is ConfigEntryState.SETUP_RETRY
+
+
+async def test_send_message_with_ttl(
+    hass: HomeAssistant, mock_pushover: MagicMock, mock_send_message: MagicMock
+) -> None:
+    """Test sending a message with TTL included in the data."""
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            "name": "pushover",
+            "api_key": "API_KEY",
+            "user_key": "USER_KEY",
+        },
+    )
+    entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    await hass.services.async_call(
+        "notify",
+        "pushover",
+        {"message": "Hello TTL", "data": {"ttl": 900}},
+        blocking=True,
+    )
+
+    assert mock_send_message.called
+    call_args = mock_send_message.call_args[0]
+    assert call_args[1] == "Hello TTL"
+    assert call_args[14] == 900
```

Comment:
Let's check all arguments
```suggestion
    mock_send_message.assert_called_once_with(user="USER_KEY", ..., ttl=900)
```